src/include/cell_inline.h

```diff
@@ -953,12 +953,16 @@ __cell_addr_window_cleanup(WT_SESSION_IMPL *session, WT_CELL_UNPACK_ADDR *unpack
     if (unpack_addr != NULL) {
         ta = &unpack_addr->ta;
         if (ta->newest_txn != WT_TXN_NONE) {
-            ta->newest_txn = WT_TXN_NONE;
-            F_SET(unpack_addr, WT_CELL_UNPACK_TIME_WINDOW_CLEARED);
+            if (!F_ISSET(session, WT_SESSION_DEBUG_DO_NOT_CLEAR_TXN_ID)) {
```

Comment:
@kommiharibabu Do you have any thoughts on this approach?

Reply:
@mm-ng Yes, the approach is fine.